Makefile.images

```diff
@@ -12,8 +12,10 @@ force_image_rebuild = $(if $(shell docker images | grep '\<$(1)\W*latest\>'),,FO
 FORCE_IMAGE: ;
 
 # Dockerfile dependencies are the file and any file copied into it
-# We have to run it through a variable in order to expand any * that might be in the COPY command
-docker_deps = $(shell files="$(1) $$(grep COPY $(1) | sed 's/COPY \(.*\) .*/\1/' )" && find $${files[*]})
+# We have to run it through a variable in order to expand any * that might be
+# in the COPY command; find is used to handle directories as dependencies
+# Files copied from another image are ignored
+docker_deps = $(shell files=($(1) $$(awk '/COPY/ { if (substr($$2, 1, 7) != "--from=") { for (i = 2; i < NF; i++) { print $$i } } }' $(1))) && find $${files[*]} -type f)
```

Comment:
As if it wasn't complicated enough? ;)

Reply:
Yeah, and it doesn’t actually work in all cases (but not because of this PR) — `find` skips files which don’t exist, so container dependencies which don’t exist aren’t listed! I’ll fix that in another PR.